File: source/UIContainer.cpp
#include "UIContainer.hpp"

namespace GTA
{
	UIContainer::UIContainer() : UIRectangle()
	{
	}
	UIContainer::UIContainer(System::Drawing::Point position, System::Drawing::Size size) : UIRectangle(position, size)
	{
	}
	UIContainer::UIContainer(System::Drawing::Point position, System::Drawing::Size size, System::Drawing::Color color) : UIRectangle(position, size, color)
	{
	}

	void UIContainer::Draw()
	{
		Draw(System::Drawing::Size());
	}
	void UIContainer::Draw(System::Drawing::Size offset)
	{
		if (!this->Enabled)
		{
			return;
		}

		UIRectangle::Draw(offset);

		for each (UIElement ^item in this->Items)
		{
			item->Draw(static_cast<System::Drawing::Size>(UIRectangle::Position + offset));
		}
	}
}

File: source/scripting_v2/GTA/UIContainer.cs
//
// Copyright (C) 2015 crosire & kagikn & contributors
// License: https://github.com/scripthookvdotnet/scripthookvdotnet#license
//

using System.Collections.Generic;
using System.Drawing;

namespace GTA
{
	public class UIContainer : UIRectangle
	{
		public UIContainer() : base()
		{
		}
		public UIContainer(Point position, Size size) : base(position, size)
		{
		}
		public UIContainer(Point position, Size size, Color color) : base(position, size, color)
		{
		}

		public List<UIElement> Items { get; set; } = new();

		public override void Draw()
		{
			Draw(new Size());
		}
		public override void Draw(Size offset)
		{
			if (!Enabled)
			{
				return;
			}

			base.Draw(offset);

			foreach (UIElement item in Items)
			{
				item.Draw(new Size(base.Position + offset));
			}
		}
	}
}
